pkg/vdri/trustbloc/config/memorycacheconfig/service.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package memorycacheconfig
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/trustbloc/trustbloc-did-method/pkg/vdri/trustbloc/models"
+
+	"github.com/bluele/gcache"
+)
+
+type config interface {
+	GetConsortium(string, string) (*models.ConsortiumFileData, error)
+	GetStakeholder(string, string) (*models.StakeholderFileData, error)
+}
+
+// ConfigService fetches consortium and stakeholder configs using a wrapped config service, caching results in-memory
+type ConfigService struct {
+	config config
+	cCache gcache.Cache
+	sCache gcache.Cache
+}
+
+// NewService create new ConfigService
+func NewService(config config) *ConfigService {
+	configService := &ConfigService{
+		config: config,
+	}
+
+	configService.cCache = gcache.New(0).LoaderExpireFunc(func(key interface{}) (interface{}, *time.Duration, error) {
+		//	TODO: fetch and return the config and its lifetime
+		keyStr, ok := key.(string)
+		if !ok {
+			return nil, nil, fmt.Errorf("key must be string")
+		}
+		return configService.getNewConsortium(keyStr, keyStr)
+	}).EvictedFunc(func(key interface{}, value interface{}) {
```

Comment:
You don't need an EvictedFunc if the cache size is 0 since the item will never be evicted.

Reply:
They are evicted if they are expired when they are retrieved with Get(). I traced the code to verify - if it didn't work this way then I wouldn't be able to use gcache for this cache.

Reply:
The expired cache entry is removed [here](https://github.com/bluele/gcache/blob/master/simple.go#L134) and that calls the EvictedFunc, if present, [here](https://github.com/bluele/gcache/blob/master/simple.go#L212).

Reply:
I see. But I don't think you need to take any action when an item is evicted since the next time Get is called, it will call your LoaderExpireFunc, and that will load another entry.

Reply:
That's true - I wanted to use the eviction callback because it's helpful scaffolding for later work where we need to verify the replacement value against the expired one - either they're the same, or the replacement has a chain of endorsements tracing back to the expired one.

I could take that out of this PR though.

Reply:
Yeah, It's better to only add the code you need to a PR.

Reply:
Done.